tests/base.py

```diff
@@ -635,3 +659,7 @@ def _sigintHandler(*args):
 
 
 signal.signal(signal.SIGINT, _sigintHandler)
+# If we insist on test databases not existing when we start, make sure we
+# check right away.
+if 'newdb' in os.environ.get('EXTRADEBUG', '').split():
```

Comment:
Should we have a separate env variable for keepdb/newdb, rather than loading extra options into `EXTRADEBUG`? Maybe something like `KEEP_TEST_DATABASES`?